puppet/chassis.rb

```diff
@@ -20,17 +20,28 @@ def self.get_extension_config(extension)
 		end
 	end
 
-	def self.get_extensions(version = nil)
-		all = Dir.glob(@@extension_dir + '/*').map { |directory| File.basename( directory ) }
+	def self.get_extensions_for_dir(ext_dir, version = nil)
+		all = Dir.glob(ext_dir + '/*').map { |directory| File.basename( directory ) }
+
+		# Remove dummy _global if found
+		all.delete("_global")
 
 		return all if ! version
 
 		all.select { |extension|
-			config = get_extension_config(extension)
+			config = get_extension_config(extension, ext_dir)
 			config['version'] == version
 		}
 	end
 
+	def self.get_extensions(version = nil)
+		self.get_extensions_for_dir(@@extension_dir, version)
+	end
+
+	def self.get_global_extensions(version = nil)
```

Comment:
I don't think you're calling this function anywhere at the moment?

Reply:
Yeah, I added this for extensions that want to use it, but we don't use it internally.